[Label Redesign]Added junction table and regression tests

panosfol opened the pull request.

panosfol committed:
Added junction table and the appropriate regression tests

A junction table was added to speed up the process of
searching in a graph. The junction table has only "id"
and "label_id" as columns.
In order for the junction table to have access to all the
vertices, it is created before the "_ag_label_vertex" table
and it is a parent to it.
In order to have the columns correctly inherited and in
the right order, the junction table is created with
"id", "properties" and "label_id" columns, but the
"properties" column is getting dropped after the creation
and inheritance of the "_ag_label_vertex".

Signed-off-by: Panagiotis Foliadis <pfoliadis@hotmail.com>

rafsun42 commented:
@panosfol Why don't your commit has any message explaining and justifying the changes you made??

panosfol commented:
> @panosfol Why don't your commit has any message explaining and justifying the changes you made??

Ill fix it

rafsun42 commented:
@panosfol Why is the label table inheriting the junction table?

panosfol commented:
@rafsun42 only the `_ag_label_vertex` inherits the junction table, because the junction table needs to have access to all the existing vertices in the graph, therefore it has one child (the `_ag_label_vertex`) and that child has all the vertex labels as children.

panosfol committed:
Fixed 'id' column of _ag_junction_table to have index
Fixed format and added some cases in the regression test

This commit focuses on three things:

- The 'id' column of _ag_junction_table was changed to
  primary key, since postgres automatically creates
  btree index for primary-key columns
- Fixed the format according the official coding
  standards of apacheAGE
- Added some more cases in the regression tests.

Signed-off-by: Panagiotis Foliadis <pfoliadis@hotmail.com>

panosfol committed:
Fix creation and usage of function table

This commit focuses on refactoring the creation and
usage of the junction table by making the following
changes :
    - The junction table doesn't utilize the inheritance
      feature and instead has a foreign-key constraint on
      its "id" column, referencing the id column of the
      "_ag_label_vertex" tablle.
    - Each vertex is both inserted in the junction table
      and the appropriate label table, therefore functions
      were added to transform and add the nodes to the
      cypher path.
    - Changes were made so the path can correctly reference
      the vertex, the junction table vertex and the edge
      respectively.
    - At the executor state there has been some small changes
      to accomodate the change of the cypher path, and to
      insert the vertex in both the junction table and the
      appropriate label table.

Signed-off-by: Panagiotis Foliadis <pfoliadis@hotmail.com>

rafsun42 commented:
Junction table may not be implemented.
